requirements.in

```diff
@@ -0,0 +1,5 @@
+pyyaml
```

Comment:
does this change in how the requirements are structured get tested in the docs ci? Will it impact running install.sh?

Reply:
This will change nothing other than the fact that by compiling our requirements, we lock versions and have explicit transitive dependencies. Downstream libraries will continue to interact with *requirements.txt* as normal.

It would be the same as if we just maintained *requirements.txt* with correct versions, this just makes it easier to maintain and upgrade. Both *requirements.in* and *requirements.txt* should be committed to version control.